test/tests/kernels/ad_mat_reaction_flexible/ad_mat_reaction_flexible_test.i

```diff
@@ -0,0 +1,88 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 10
+  ny = 10
+[]
+
+[Variables]
+  [./c_a]
+  [../]
+  [./c_b]
+  [../]
+[]
+
+[ICs]
+  [c_a_IC]
+    type = ConstantIC
+    variable = c_a
+    value = 1
+  []
+  [c_b_IC]
+    type = ConstantIC
+    variable = c_b
+    value = 1
+  []
+[]
+
+[Kernels]
+  [./timeDerivative_c_a]
+    type     = ADTimeDerivative
+    variable = c_a
+  [../]
+  [./timeDerivative_c_b]
+    type     = TimeDerivative
+    variable = c_b
+  [../]
+  [./MatReaction]
+    type     = ADMatReactionFlexible
+    variable = c_b
+    vs = 'c_a'
+    coeff = 0.5
+    mob_name = K
+  [../]
+[]
+
+[Materials]
+  [./K]
+    type = ADParsedMaterial
+    f_name = 'K'
+    function = '10'
+  [../]
+[]
+
+[BCs]
+  [./c_a_neumann] # No flux on the sides
+    type = NeumannBC
+    variable = c_a
+    boundary = 'left right bottom top'
+    value = 0
+  [../]
+  [./c_b_neumann] # No flux on the sides
+    type = NeumannBC
+    variable = c_b
+    boundary = 'left right bottom top'
+    value = 0
+  [../]
+[]
+
+[Executioner]
+  type                 = Transient
+  scheme               = bdf2
+  nl_rel_tol           = 1e-10
+
+  solve_type = 'PJFNK'
```

Comment:
Since we are using AD for this new object, let's do `solve_type = NEWTON`. If we have a perfect jacobian, NEWTON will always be better unless we have memory concerns due to a high number of nonlinear variables (which we don't have here)

Reply:
You're absolutely right! I copy/pasted another input without thinking about that part of the file. Thank you for your careful review!